### ISSUE TITLE
AR teardown can close Session while GL thread is still using it

### ISSUE THREAD

**Issue by HereLiesAz** [HereLiesAz/GraffitiXR#1910]
## Problem

`ArRenderer.detachSessionBounded()` is used as a synchronization barrier before off-GL-thread ARCore session work, but when its `sessionLock.tryLock()` times out it still nulls the renderer's volatile session reference and returns `false` while the in-flight GL frame may retain a local reference to the same native `Session`.

Two production callers then fail open:

### Full cleanup

```kotlin
renderer?.isDestroying = true
val glThreadOut = renderer?.detachSessionBounded(1500L) ?: true
if (!glThreadOut) {
    appendDiag("cleanup: GL thread wedged in-frame after 1500ms — closing session anyway")
}
...
session?.let {
    if (isSessionResumed) it.pause()
    it.close()
}
```

If the lock timed out, the GL thread may still be inside `Session.update()` or another ARCore call while cleanup deliberately calls `pause()` / `close()` underneath it.

### Stereo-recovery reconfiguration

The same helper is used before live camera reconfiguration. On timeout that caller logs `reconfiguring anyway`, then calls the pause/configuration path on the same session while the renderer may still be using it. This is the same ownership violation even though it does not necessarily close the session.

ARCore's `Session.close()` contract explicitly says it is **not safe to call methods on the session or objects obtained from it while the session is being closed**. More generally, the repo already treats `Session`/`Anchor` access as requiring serialization against the render loop.

Official reference: https://developers.google.com/ar/reference/java/com/google/ar/core/Session#close()

## History

The bounded-detach / fail-open behavior dates to commit `ec24c9b54656841cd16bb6a8139cf08d85fbfeed` (2026-06-11), so this is a latent lifecycle defect rather than a regression introduced by the September ARCore 1.56 upgrade. Runtime/timing changes can make it easier to expose.

The ARCore 1.56 adaptation in PR #1909 fixes a related but separate race: `AnchorOrchestrator.clear()` becomes local-state-only and no longer makes `Anchor.detach()` native calls after a failed renderer-lock acquisition. It does **not** make concurrent `Session.close()` or reconfiguration safe.

## Required invariant

`detachSessionBounded()` must be a real ownership handoff, not merely a best-effort volatile null:

- if it reports success, the renderer is out of the session critical section and the caller may take ownership;
- if it reports timeout, the caller must not pause, configure, close, or otherwise touch that session concurrently with the in-flight renderer call;
- setting `isDestroying = true` may stop *new* frames, but it does not invalidate the current frame's local session reference.

## Recovery shape

UI/main-thread exit must remain bounded. A wedged GL frame should therefore use an explicit deferred cleanup/handoff path that acquires the renderer's session lock before native pause/close, rather than blocking the UI indefinitely or racing close under use. If the native call never returns, leaking/abandoning that stuck session is safer than invoking undefined concurrent native teardown.

Live reconfiguration must fail/abort cleanly on timeout rather than `reconfiguring anyway`; it can retry later once ownership is actually obtained.

## Acceptance criteria

- cleanup never calls `Session.pause()`/`close()` concurrently with an in-flight renderer ARCore call
- stereo/camera reconfiguration never mutates a session after renderer handoff timed out
- `detachSessionBounded()` does not pretend to detach by nulling `session` off-lock on timeout
- exit remains bounded from the UI/main thread
- a wedged renderer has an explicit deferred or abandoned-session recovery path rather than close-under-use
- timeout behavior has regression coverage at the ownership-decision level
- preserve existing NavHost/AzNavRail fixes and ARCore 1.56.0

**Comment by HereLiesAz** [comment]
🤖 Hi @HereLiesAz, Jules is on it. Track progress in [the logs](https://github.com/HereLiesAz/GraffitiXR/actions/runs/34791253444).
